dsm/resource_aws_sobject.go

```diff
@@ -303,14 +308,11 @@ func resourceCreateAWSSobject(ctx context.Context, d *schema.ResourceData, m int
 		}
 	}
 
-	req, err := m.(*api_client).APICallBody("POST", endpoint, security_object)
+	req, err := invokeAWSCreateAPI(m, security_object, endpoint)
+	// Irrespective of the creation status, lock will be released.
```

Comment:
Since the critical section is `invokveAWSCreateAPI`, it might be better to move the `aws_sobject_lock.Unlock()` inside the function body before returning any errors/results. 

Right now it could be unclear at a glance why the release of lock is here when lines prior it wasn't locked.

Reply:
This makes sense. I addressed accordingly.